### PR TITLE
Planning: initial implementation for lane-change urgency checking

### PR DESCRIPTION
**Pull request by JasonXu12**
Current logic is simply stopping and waiting. Future implementation will be based on this.

### CODE REVIEW

File: modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc [hunk 1]
```diff
@@ -159,6 +164,51 @@ Status SpeedBoundsDecider::Process(
...
+              if (distance_to_passage_end < \
+                  speed_bounds_config_.approach_distance_for_lane_change()) {
+                // Set a stop fence and wait for chances to change lane
+                const std::string stop_wall_id = "lane_change_stop";
+                std::vector<std::string> wait_for_obstacles;
+                DeciderRuleBasedStop::BuildStopDecision(
+                       stop_wall_id, sl_point.s(),
+                       speed_bounds_config_.urgent_distance_for_lane_change(),
+                       StopReasonCode::STOP_REASON_LANE_CHANGE_URGENCY,
+                       wait_for_obstacles, frame, &reference_line_info);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc]: Try to reduce indents. Some practices:

1. You can check the negative logic and call `continue`.
```
if (is_clear_to_change_lane_) {
  continue;
}
// Other logic.
```

2. You can wrap inside logic into function to make it more readable:
```
if (!is_clear_to_change_lane_) {
  DealWithIt();
}
```

File: modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc [hunk 2]
```diff
@@ -159,6 +164,51 @@ Status SpeedBoundsDecider::Process(
...
+        is_clear_to_change_lane_ = \
+            ChangeLaneDecider::IsClearToChangeLane(&reference_line_info);
+      } else {
+        // If lane change is blocked
+        if (!is_clear_to_change_lane_) {
+          // Get the end point of current routing
+          const auto& route_end_waypoint = \
+          reference_line_info.Lanes().RouteEndWaypoint();
+          if (route_end_waypoint.lane) {
+            auto point = \
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc]: No need to use `\` in C++.

File: modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc (hunk 1 above)
Comment (JasonXu12) [reply modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc]: Solved. Thanks!

File: modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc (hunk 2 above)
Comment (JasonXu12) [reply modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc]: Solved. Thanks!

**Pull request merged**